tests/unit/processing/test_handle_error.py

```diff
@@ -0,0 +1,360 @@
+from backend.layers.processing.upload_failures.app import get_failure_slack_notification_message, parse_event
+
+
+@pytest.fixture
+def sample_slack_header_block():
+    return {
+        "type": "header",
+        "text": {
+            "type": "plain_text",
+            "text": "Dataset failed to process:fire:",
+            "emoji": True,
+        },
+    }
+
+
+@pytest.fixture
+def sample_slack_status_block():
+    return {
+        "type": "section",
+        "text": {
+            "type": "mrkdwn",
+            "text": "```{\n"
+            '  "cxg_status": null,\n'
+            '  "h5ad_status": null,\n'
+            '  "processing_status": null,\n'
+            '  "rds_status": null,\n'
+            '  "upload_status": null,\n'
+            '  "validation_message": null,\n'
+            '  "validation_status": null\n'
+            "}```",
+        },
+    }
+
+
+@pytest.fixture
+def sample_slack_status_block_empty():
+    return {
+        "type": "section",
+        "text": {
+            "type": "mrkdwn",
+            "text": "``````",
+        },
+    }
+
+
+def test_parse_event_with_empty_event():
+    (
+        dataset_id,
+        collection_version_id,
+        error_step_name,
+        error_job_id,
+        error_aws_regions,
+        error_cause,
+        execution_arn,
+    ) = parse_event({})
+
+    assert dataset_id is None
+    assert collection_version_id is None
+    assert error_step_name is None
+    assert error_job_id is None
+    assert error_aws_regions is None
+    assert error_cause == ""
+    assert execution_arn is None
+
+
+def test_parse_event_with_error_cause():
+    error_cause = (
+        '{"JobName": "Step1", "JobId": "789", "Container": {"Environment": [{"Name": "AWS_DEFAULT_REGION", '
+        '"Value": "us-east-1"}]}}'
+    )
+    event = {"execution": "arn", "dataset_id": "123", "collection_id": "456", "error": {"Cause": error_cause}}
+    (
+        dataset_id,
+        collection_version_id,
+        error_step_name,
+        error_job_id,
+        error_aws_regions,
+        error_cause,
+        execution_arn,
+    ) = parse_event(event)
+
+    assert dataset_id == "123"
+    assert collection_version_id == "456"
+    assert error_step_name == "Step1"
+    assert error_job_id == "789"
+    assert error_aws_regions == "us-east-1"
+    assert error_cause is error_cause
+    assert execution_arn == "arn"
+
+
+def test_parse_event_without_error_cause():
+    event = {"dataset_id": "123", "collection_id": "456", "error": {}}
+
+    (
+        dataset_id,
+        collection_version_id,
+        error_step_name,
+        error_job_id,
+        error_aws_regions,
+        error_cause,
+        execution_arn,
+    ) = parse_event(event)
+
+    assert dataset_id == "123"
+    assert collection_version_id == "456"
+    assert error_step_name is None
+    assert error_job_id is None
+    assert error_aws_regions is None
+    assert error_cause == ""
+    assert execution_arn is None
```

Comment:
isn't convention to use the unittest framework to make these assertions via `self.assertIsNone`, etc.

is there a reason you decided to structure these tests differently rather than create a class and inherit from `unittest.TestCase`?

Reply:
We've been moving to use pytest when possible for a few reason, but I think this biggest reason is it has better support for reporting. Since this was a new set of tests, there was no reason to keep it as a `unittest.TestCase`

Reply:
I've also found that pytest has trouble running unittests with subtests. Using pytest test format avoid this problem.

Reply:
I see. And pytest does not provide any proprietary assertion statements. Got it.

Reply:
>We've been moving to use pytest when possible for a few reason 

in summary what were these? I think I missed this discussion / decision

Reply:
simplified test structure, better reporting, already being used in other single-cell projects.